core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -384,6 +352,33 @@ class UserService(protected val userInfo: UserInfo, val dataSource: SlickDataSou
     }
   }
 
+  private def googleRoleWhitelistCheck(role: String): Future[Unit] = {
+    if (projectOwnerGrantableRoles.contains(role))
+      Future.successful(())
+    else
+      Future.failed(new RawlsExceptionWithErrorReport(errorReport = ErrorReport(StatusCodes.Forbidden, s"Cannot alter Google role $role: not in list [${projectOwnerGrantableRoles mkString ", "}]")))
+  }
+
+  def grantGoogleRoleToUser(projectName: RawlsBillingProjectName, targetUserEmail: WorkbenchEmail, role: String): Future[PerRequestMessage] = {
+    for {
+      _ <- googleRoleWhitelistCheck(role)
+      proxyGroupEmail <- samDAO.getProxyGroup(userInfo, targetUserEmail)
```

Comment:
what happens if someone mistakenly puts a) a pet email b) the proxy group in here?

Reply:
Sam looks up the user subject ID for the email, so Proxy Group and Pet would both get 404